SimulationAgent/DeviceTelemetry/SendTelemetry.cs

```diff
@@ -35,30 +35,40 @@ public void Init(IDeviceTelemetryActor context, string deviceId, DeviceModel dev
 
         public async Task RunAsync()
         {
-            var state = this.context.DeviceState.GetAll();
-
-            // device could be rebooting, updating firmware, etc.
-            this.log.Debug("Checking to see if device is online", () => new { this.deviceId });
-            if (!state.ContainsKey("online") || (bool) state["online"])
+            try
             {
-                this.log.Debug("The device state says the device is online, sending telemetry...", () => new { this.deviceId });
+                var state = this.context.DeviceState.GetAll();
+
+                // device could be rebooting, updating firmware, etc.
+                this.log.Debug("Checking to see if device is online", () => new { this.deviceId });
+                if (!state.ContainsKey("online") || (bool)state["online"])
```

Comment:
Formatting (spaces)

Reply:
This is existing code. Added a space though